sudospawner/spawner.py

```diff
@@ -43,9 +43,15 @@ def do(self, action, **kwargs):
         yield p.stdin.write(json.dumps(kwargs).encode('utf8'))
         p.stdin.close()
         data = yield p.stdout.read_until_close()
+        # Read again if output is not json
+        if not data.decode('utf8').startswith('{'):
+            data = yield p.stdout.read_until_close()
```

Comment:
I really don't know. I came up with this based on my understanding of the yield keyword. 

What I observed was that if pam_oddjob_mkhomedir ran, the only thing captured by read_until_close would be pam_oddjob_mkhomedir's output `Creating home directory for [user].`. Calling read_until_close again resulted in `{"pid": 6322}Creating home directory for [user].`.